### PR TITLE
20 us as a user i want to play the pong game in real time

### PR DESCRIPTION
**Pull request by COTHSC**
# Description

Please provide a detailed description of what was done in this PR.
Precise the issue that you are resolving.

# Changes include

- [x] New feature (non-breaking change that adds functionality)
- [ ] Bugfix (non-breaking change that solves an issue)

# Breaking changes

Please complete this section if any breaking changes have been made, otherwise delete it.

# Checklist

- [x] I have assigned this PR to myself
- [x] I have added at least 1 reviewer
- [x] I have added the needed labels
- [x] I have linked this PR to an issue
- [x] I have linked this PR to a milestone
- [x] I have linked this PR to a project
- [x] I have tested this code
- [ ] I have added / updated tests (unit / functionals / end-to-end / ...)
- [ ] I have updated the README and other relevant documents (guides...)
- [ ] I have added sufficient documentation both in code, as well as in the READMEs

# Additional comments

Please post additional comments in this section if you have them, otherwise delete it.

### CODE REVIEW

File: backend/app/src/pong/entities/position.entity.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
+export class Position {
```
Comment (cnstll) [inline backend/app/src/pong/entities/position.entity.ts:1]: SHould be interfaces right ?

File: backend/app/src/pong/game.gateway.ts [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+import { SchedulerRegistry } from '@nestjs/schedule';
+import { JwtAuthGuard } from 'src/auth/guard/jwt.auth-guard';
+import { UseGuards } from '@nestjs/common';
+import { Game, Status } from './entities/position.entity';
+import { GetCurrentUserId } from 'src/common/decorators/getCurrentUserId.decorator';
+
+@WebSocketGateway({
+  cors: {
+    origin: [
+      'http://localhost',
```
Comment (cnstll) [inline backend/app/src/pong/game.gateway.ts:20]: Do we need all these origins ? 
origin: "http://localhost:8080" should be enough no ?

File: backend/app/src/strategy/jwt.strategy.ts [hunk 3]
```diff
@@ -19,6 +20,18 @@ export class JwtStrategy extends PassportStrategy(Strategy, 'jwt') {
     return jwtPayload;
   }
 
+  private static extractJWTfromHandshake(req: any): string | null {
```
Comment (cnstll) [inline backend/app/src/strategy/jwt.strategy.ts:23]: type with Express ?

**Review by cnstll — CHANGES_REQUESTED** [review]
LNAGTM

File: backend/app/src/pong/game.gateway.ts (hunk 2 above)
Comment (cnstll) [reply backend/app/src/pong/game.gateway.ts:20]: or just "http://localhost" ?

File: backend/app/src/pong/entities/position.entity.ts (hunk 1 above)
Comment (cnstll) [reply backend/app/src/pong/entities/position.entity.ts:1]: @COTHSC 
"When to use classes: When you want to create objects that have actual function code in them, have a constructor for initialization, and/or you want to create instances of them with new. Also, for simple data objects, you can use classes for setting up default values. Another time you would want to use them is when you are doing type chec" https://stackoverflow.com/questions/40973074/difference-between-interfaces-and-classes-in-typescript

Comment (COTHSC) [reply backend/app/src/pong/entities/position.entity.ts]: Resolved!

File: backend/app/src/strategy/jwt.strategy.ts (hunk 3 above)
Comment (COTHSC) [reply backend/app/src/strategy/jwt.strategy.ts:23]: Can't type with express, we would need to do a custom type, with not very much added value, I'll do it to get rid of the warning

**Review by llescure — APPROVED** [review]
🕺

File: backend/app/src/strategy/jwt.strategy.ts (hunk 3 above)
Comment (COTHSC) [reply backend/app/src/strategy/jwt.strategy.ts]: RESOLVED!

File: backend/app/src/pong/game.gateway.ts (hunk 2 above)
Comment (COTHSC) [reply backend/app/src/pong/game.gateway.ts]: This. is. resolved.

**Review by cnstll — APPROVED** [review]

**Pull request merged**